src/types/host.ts

```diff
@@ -74,3 +77,30 @@ export enum HostMonitorOp {
   ProvisionFailed = "provision_failed",
   Expired = "expired",
 }
+
+export enum HostsTableFilterParams {
+  CurrentTaskId = "currentTaskId",
+  DistroId = "distroId",
+  HostId = "hostId",
+  StartedBy = "startedBy",
+  Statuses = "statuses",
+}
+
+export const mapQueryParamToId: Record<HostsTableFilterParams, HostSortBy> = {
+  [HostsTableFilterParams.HostId]: HostSortBy.Id,
+  [HostsTableFilterParams.DistroId]: HostSortBy.Distro,
+  [HostsTableFilterParams.Statuses]: HostSortBy.Status,
+  [HostsTableFilterParams.CurrentTaskId]: HostSortBy.CurrentTask,
+  [HostsTableFilterParams.StartedBy]: HostSortBy.Owner,
+} as const;
+
+export const mapIdToFilterParam: PartialRecord<
+  HostSortBy,
+  keyof HostsQueryVariables
+> = Object.entries(mapQueryParamToId).reduce(
+  (obj, [id, param]) => ({
+    ...obj,
+    [param]: id,
+  }),
+  {}
```

Comment:
I'm confused by this what are we trying to do here?
Is this trying to establish a mapping between the query param and the react-table id field?

Reply:
Yep that's it. More specifically, a given field has a value that is used with the `sortBy` param, as well as a field that is used as the key when filtering (i.e. `distroId` in `distroId=ubuntu`). The goal here is to associate them with each other and their column.